### ISSUE TITLE
DialogFlow : A complete guide with webhook

### ISSUE THREAD

**Issue by Depado** [Depado/articles#26]
This is the comment issue for the [DialogFlow : A complete guide with webhook](https://blog.depado.eu/post/dialogflow-golang-webhook) article.

**Comment by M6Jesus** [comment]
please, why can't find examples with java language?? its very difficult to have documentation in java :( could you help ou give a link? please

**Comment by Depado** [comment]
Hi
There's an official [Java SDK](https://github.com/GoogleCloudPlatform/google-cloud-java/tree/master/google-cloud-clients/google-cloud-dialogflow) for DialogFlow with a [Quickstart](https://github.com/GoogleCloudPlatform/google-cloud-java/tree/master/google-cloud-clients/google-cloud-dialogflow#quickstart) section.

Otherwise this should look quite identitcal to what I'm showing in this article, given that you know how to handle HTTP requests with Java. I'm sorry I don't have much knowledge in Java unfortunately.

**Comment by M6Jesus** [comment]
thanks for you response. I dont think that i need to use SDK, it's just an
asignement to build a java webhook who can make google home to give
different answers from my API? :(

2018-06-04 11:02 GMT+02:00 Depado <notifications@github.com>:

**Comment by Depado** [comment]
> @M6Jesus
> thanks for you response. I dont think that i need to use SDK, it's just an
> asignement to build a java webhook who can make google home to give
> different answers from my API? :(

The SDK may come in handy, at least for the different types and context manipulation. Otherwise it's a simple question of receiving the context sent to your webhook, and sending back a response that Dialogflow can understand.

**Comment by M6Jesus** [comment]
exactly, I just want to send string data to dialogflow so that google home
could pronouced it! but i don't know how
in the documentation they said that we just have to passed  directly the
value to  QueryResult.fulfillment_text. but how can i get the QueryREsult?
i search and find this classes

import com.google.cloud.dialogflow.v2beta1.QueryResult;
import com.google.cloud.dialogflow.v2beta1.WebhookRequest;
import com.google.cloud.dialogflow.v2beta1.WebhookResponse;

so i try to use this queryResult first by doinng this in my webhook

@PostMapping("/message3")
    public WebhookResponse essai3(WebhookRequest request) {
        System.out.println(request.toString());
        QueryResult q = request.getQueryResult();

        return WebhookResponse.newBuilder().setFulfillmentText("debrouille
toi a me repondre").build();
    }

but i have an error of parsing and non existing class..

I don't know if i have to use google.cloud

Sorry i'm just a student and newer. sorry also for my english i'm
francophone :)

thank a lot for your help

2018-06-04 11:27 GMT+02:00 Depado <notifications@github.com>:

**Comment by nbfarrand** [comment]
This example is very helpful.  It is a similar use case I am working on.  Would you be able to checkin a zip of the DialogFlow agent into GitHub.  I have a few questions based on your article around the DiaglogFlow intents.  Regardless, this is a very helpful article.

**Comment by Depado** [comment]
> @nbfarrand
> This example is very helpful.  It is a similar use case I am working on.  Would you be able to checkin a zip of the DialogFlow agent into GitHub?  I have a few questions based on your article around the DiaglogFlow intents.  Regardless, this is a very helpful article.

I could do that but I'm not sure the agent is complete. You can find me on IRC (freenode), [Twitter](https://twitter.com/Depado_), [Keybase](https://keybase.io/depado), etc. Feel free to message me !
